### ISSUE TITLE
Cant install on windows 8.1

### ISSUE THREAD

**Issue by teebu** [dachev/node-cld#32]
npm install cld

gets a bunch of warnings and here's the part where it fails:
```
cld_generated_score_quad_octa_0122.obj : warning LNK4006: "short const * const CLD2::kAvgDeltaOctaScore" (?kAvgDeltaOctaScore@CLD2@@3QBFB) already defined in cld_generated_score_quad_octa_2.obj; second definition ignored [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\deps\cld\cld-c .vcxproj]
cld_generated_score_quad_octa_0122.obj : warning LNK4006: "int const CLD2::kAvgDeltaOctaScoreSize" (?kAvgDeltaOctaScoreSize@CLD2@@3HB) already defined in cld_generated_score_quad_octa_2.obj; second definition ignored [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\deps\cld\cld-c.vcx proj]
cld2_generated_quadchrome_16.obj : warning LNK4006: "struct CLD2::CLD2TableSummary const CLD2::kQuad_obj" (?kQuad_obj@CLD2@@3UCLD2TableSummary@1@B) already defined in cld2_generated_quadchrome_2.obj; second definition ignored [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\deps\cld\ cld-c.vcxproj]
cld2_generated_quadchrome_16.obj : warning LNK4006: "struct CLD2::CLD2TableSummary const CLD2::kQuad_obj2" (?kQuad_obj2@CLD2@@3UCLD2TableSummary@1@B) already defined in cld2_generated_quadchrome_2.obj; second definition ignored [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\deps\cl d\cld-c.vcxproj]
cld2_generated_distinctocta0122.obj : warning LNK4006: "struct CLD2::CLD2TableSummary const CLD2::kDistinctOcta_obj2" (?kDistinctOcta_obj2@CLD2@@3UCLD2TableSummary@1@B) already defined in cld2_generated_distinctoctachrome.obj; second definition ignored [C:\Users\homer\Documents\Learning\cld\node_ modules\cld\build\deps\cld\cld-c.vcxproj]
cld2_generated_distinctocta0122.obj : warning LNK4006: "unsigned int const CLD2::kDistinctOcta2IndSize" (?kDistinctOcta2IndSize@CLD2@@3IB) already defined in cld2_generated_distinctoctachrome.obj; second definition ignored [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\deps\cld\cld -c.vcxproj]
cld2_generated_distinctocta0122.obj : warning LNK4006: "unsigned int const CLD2::kDistinctOctaIndSize" (?kDistinctOctaIndSize@CLD2@@3IB) already defined in cld2_generated_distinctoctachrome.obj; second definition ignored [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\deps\cld\cld-c .vcxproj]
cld2_generated_distinctocta0122.obj : warning LNK4006: "struct CLD2::CLD2TableSummary const CLD2::kDistinctOcta_obj" (?kDistinctOcta_obj@CLD2@@3UCLD2TableSummary@1@B) already defined in cld2_generated_distinctoctachrome.obj; second definition ignored [C:\Users\homer\Documents\Learning\cld\node_mo dules\cld\build\deps\cld\cld-c.vcxproj]
cld2_generated_deltaocta0122.obj : warning LNK4006: "struct CLD2::CLD2TableSummary const CLD2::kDeltaOcta_obj2" (?kDeltaOcta_obj2@CLD2@@3UCLD2TableSummary@1@B) already defined in cld2_generated_deltaoctachrome.obj; second definition ignored [C:\Users\homer\Documents\Learning\cld\node_modules\cld\ build\deps\cld\cld-c.vcxproj]
cld2_generated_deltaocta0122.obj : warning LNK4006: "unsigned int const CLD2::kDeltaOcta2IndSize" (?kDeltaOcta2IndSize@CLD2@@3IB) already defined in cld2_generated_deltaoctachrome.obj; second definition ignored [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\deps\cld\cld-c.vcxproj]
cld2_generated_deltaocta0122.obj : warning LNK4006: "struct CLD2::CLD2TableSummary const CLD2::kDeltaOcta_obj" (?kDeltaOcta_obj@CLD2@@3UCLD2TableSummary@1@B) already defined in cld2_generated_deltaoctachrome.obj; second definition ignored [C:\Users\homer\Documents\Learning\cld\node_modules\cld\bu ild\deps\cld\cld-c.vcxproj]
cld2_generated_deltaocta0122.obj : warning LNK4006: "unsigned int const CLD2::kDeltaOctaIndSize" (?kDeltaOctaIndSize@CLD2@@3IB) already defined in cld2_generated_deltaoctachrome.obj; second definition ignored [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\deps\cld\cld-c.vcxproj]
cld2_generated_quad0122.obj : warning LNK4006: "struct CLD2::CLD2TableSummary const CLD2::kQuad_obj" (?kQuad_obj@CLD2@@3UCLD2TableSummary@1@B) already defined in cld2_generated_quadchrome_2.obj; second definition ignored [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\deps\cld\cld-c .vcxproj]
cld2_generated_quad0122.obj : warning LNK4006: "struct CLD2::CLD2TableSummary const CLD2::kQuad_obj2" (?kQuad_obj2@CLD2@@3UCLD2TableSummary@1@B) already defined in cld2_generated_quadchrome_2.obj; second definition ignored [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\deps\cld\cld -c.vcxproj]
  cld-c.vcxproj -> C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\Release\\cld-c.lib
  constants.cc
  cld.cc
  win_delay_load_hook.cc
C:\Program Files (x86)\Windows Kits\10\Include\10.0.10240.0\ucrt\corecrt_terminate.h(33): error C2872: 'terminate_handler': ambiguous symbol (compiling source file ..\src\cld.cc) [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\cld.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.10240.0\ucrt\corecrt_terminate.h(18): note: could be 'void (__cdecl *__cdecl terminate_handler)(void)' (compiling source file ..\src\cld.cc)
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\exception(192): note: or       'std::terminate_handler' (compiling source file ..\src\cld.cc)
C:\Program Files (x86)\Windows Kits\10\Include\10.0.10240.0\ucrt\corecrt_terminate.h(34): error C2872: 'terminate_handler': ambiguous symbol (compiling source file ..\src\cld.cc) [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\cld.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.10240.0\ucrt\corecrt_terminate.h(18): note: could be 'void (__cdecl *__cdecl terminate_handler)(void)' (compiling source file ..\src\cld.cc)
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\exception(192): note: or       'std::terminate_handler' (compiling source file ..\src\cld.cc)
C:\Program Files (x86)\Windows Kits\10\Include\10.0.10240.0\ucrt\corecrt_terminate.h(37): error C2872: 'terminate_handler': ambiguous symbol (compiling source file ..\src\cld.cc) [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\cld.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.10240.0\ucrt\corecrt_terminate.h(18): note: could be 'void (__cdecl *__cdecl terminate_handler)(void)' (compiling source file ..\src\cld.cc)
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\exception(192): note: or       'std::terminate_handler' (compiling source file ..\src\cld.cc)
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\eh.h(35): error C2872: 'unexpected_handler': ambiguous symbol (compiling source file ..\src\cld.cc) [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\cld.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\eh.h(19): note: could be 'void (__cdecl *__cdecl unexpected_handler)(void)' (compiling source file ..\src\cld.cc)
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\exception(193): note: or       'std::unexpected_handler' (compiling source file ..\src\cld.cc)
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\eh.h(36): error C2872: 'unexpected_handler': ambiguous symbol (compiling source file ..\src\cld.cc) [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\cld.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\eh.h(19): note: could be 'void (__cdecl *__cdecl unexpected_handler)(void)' (compiling source file ..\src\cld.cc)
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\exception(193): note: or       'std::unexpected_handler' (compiling source file ..\src\cld.cc)
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\eh.h(39): error C2872: 'unexpected_handler': ambiguous symbol (compiling source file ..\src\cld.cc) [C:\Users\homer\Documents\Learning\cld\node_modules\cld\build\cld.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\eh.h(19): note: could be 'void (__cdecl *__cdecl unexpected_handler)(void)' (compiling source file ..\src\cld.cc)
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\exception(193): note: or       'std::unexpected_handler' (compiling source file ..\src\cld.cc)
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 6.3.9600
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\homer\Documents\Learning\cld\node_modules\cld
gyp ERR! node -v v4.4.2
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok
npm ERR! Windows_NT 6.3.9600
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "cld"
npm ERR! node v4.4.2
npm ERR! npm  v2.15.0
npm ERR! code ELIFECYCLE

npm ERR! cld@2.4.6 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the cld@2.4.6 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the cld package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs cld
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!
npm ERR!     npm owner ls cld
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\homer\Documents\Learning\cld\npm-debug.log
```

**Comment by CharlieHess** [comment]
Same thing here when building with VS2015.
```
C:\Program Files (x86)\Windows Kits\10\Include\10.0.10240.0\ucrt\corecrt_terminate.h(33): error C2872: 'terminate_handler': ambiguous symbol (compiling source file ..\src\cld.cc) [C:\Source\slack-winssb\node_modules\cld\build\cld.vcxproj]
```

**Comment by mjeries** [comment]
Same error as CharlieHess on Windows 7 using [option 2](https://github.com/nodejs/node-gyp/blob/master/README.md).

**Comment by dachev** [comment]
This should be fixed in 2.4.7
